ts/src/coinlist.ts

```diff
@@ -0,0 +1,2246 @@
+        //                 order_id: '39911d5f-c789-4a7d-ad34-820a804d1da6',
+        //                 quantity: '-0.0009',
+        //                 price: '1608.83000000',
+        //                 fee: '0.006516',
+        //                 fee_type: 'taker',
+        //                 fee_currency: 'USDT',
+        //                 logical_time: '2023-10-20T13:16:30.000Z'
+        //             },
+        //             {
+        //                 symbol: 'ETH-USDT',
+        //                 auction_code: 'ETH-USDT-2023-10-20T13:22:14.000Z',
+        //                 order_id: '83ed365f-497d-433b-96c1-9d08c1a12842',
+        //                 quantity: '0.0008',
+        //                 price: '1615.24000000',
+        //                 fee: '0.005815',
+        //                 fee_type: 'taker',
+        //                 fee_currency: 'USDT',
+        //                 logical_time: '2023-10-20T13:22:14.000Z'
+        //             },
+        //         ]
+        //     }
+        //
+        const fills = this.safeValue (response, 'fills', []);
+        return this.parseTrades (fills, market, since, limit);
+    }
+
+    async fetchOrderTrades (id: string, symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name coinlist#fetchOrderTrades
+         * @description fetch all the trades made from a single order
+         * @param {string} id order id
+         * @param {string} symbol unified market symbol
+         * @param {int} [since] the earliest time in ms to fetch trades for
+         * @param {int} [limit] the maximum number of trades to retrieve
+         * @param {object} [params] extra parameters specific to the coinlist api endpoint
+         * @returns {object[]} a list of [trade structures]{@link https://github.com/ccxt/ccxt/wiki/Manual#trade-structure}
+         */
+        const request = {
+            'order_id': id,
+        };
+        return await this.fetchMyTrades (symbol, since, limit, this.extend (request, params));
+    }
+
+    async fetchOrders (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name coinlist#fetchOrders
+         * @description fetches information on multiple orders made by the user
+         * @param {string} symbol unified market symbol of the market orders were made in
+         * @param {int} [since] the earliest time in ms to fetch orders for
+         * @param {int} [limit] the maximum number of  orde structures to retrieve (default 200, max 500)
+         * @param {object} [params] extra parameters specific to the coinlist api endpoint
+         * @returns {Order[]} a list of [order structures]{@link https://github.com/ccxt/ccxt/wiki/Manual#order-structure}
+         */
+        await this.loadMarkets ();
+        let status = this.safeString (params, 'status', undefined) as any;
```

Comment:
we don't need this, `safeString` returns undefined if the key is not found so this is enough `this.safeString (params, 'status')`